drafts/compound-components.md

```diff
@@ -0,0 +1,252 @@
+---
+title: The Art of Component Composition - Leveraging Compound Components in React
+domain: software-engineering-corner.zuehlke.com
+tags: reactjs, components, patterns
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1709986147750/6D9yz4Rsi.png?auto=format
+publishAs: gadu
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+---
+
+As frontend applications get bigger and bigger, they have to cover more and more use cases and therefore their UI components need to get more flexible. This blog post describes how the requirements for a UI component increase throughout a typical project. We show what problems this entails and how they can be solved with the compound component pattern.
+
+## Exploring the issue by the example of a dialog
+
+We have a React application that can manage files. One feature is to delete files, another is to duplicate them. For both actions, the action should be confirmed with a dialog. Since deleting is a destructive action, we also want to color the delete button red. In such a case, we want to implement a dialog component that we can use for both scenarios. We, therefore, need various parameters to configure the dialog accordingly. The following code shows how the call of such a component could look like:
```

Comment:
```suggestion
We have a React application that can manage files. One feature is to delete files, another is to duplicate them. For both actions, the action should be confirmed with a dialog. Since deleting is a destructive action, we also want to color the delete button red. In such a case, we want to implement a dialog component that we can use for both scenarios. We, therefore, need various parameters to configure the dialog accordingly. The Dialog Component could look like this:
```